src/main/java/Ui.java

```diff
@@ -0,0 +1,106 @@
+import Tasks.Deadline;
+import Tasks.Event;
+import Tasks.Task;
+import Tasks.ToDo;
+
+import java.time.LocalDateTime;
+import java.util.Scanner;
+
+public class Ui {
+
+    public TaskList run(TaskList tasks) {
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+
+        Scanner sc = new Scanner(System.in);
+
+        while (true) {
+            Parser parser = new Parser(sc.nextLine());
+
+            if (!parser.inputsAreValid()) {
+                continue;
+            }
+
+            String action = parser.getAction();
+            String description = parser.getDescription();
+            LocalDateTime by = parser.getBy();
+            LocalDateTime at = parser.getAt();
+
+            switch (action) {
+                case "bye":
```

Comment:
According to the modules' coding standards, case and default should be at the same indentation as the switch statement.